src/modules/fw_mode_manager/FixedWingModeManager.cpp

```diff
@@ -804,9 +788,18 @@ FixedwingPositionControl::control_auto_position(const float control_interval, co
 
 	_longitudinal_ctrl_sp_pub.publish(fw_longitudinal_control_sp);
 
-	_ctrl_limits_handler.setThrottleMax(throttle_max);
-	_ctrl_limits_handler.setThrottleMin(throttle_min);
-	_ctrl_limits_handler.setSpeedWeight((pos_sp_curr.gliding_enabled ? 2.0f : NAN));
+	float throttle_min = NAN;
+	float throttle_max = NAN;
+
+	if (pos_sp_curr.gliding_enabled) {
+		/* enable gliding with this waypoint */
+		throttle_min = 0.0;
+		throttle_max = 0.0;
+		_ctrl_configuration_handler.setSpeedWeight(2.f);
+	}
+
+	_ctrl_configuration_handler.setThrottleMax(throttle_max);
```

Comment:
@sfuhrer Actually, I think we can simplify this even more and just the setting of throttle min and max into the if statement above. When you don't set anything, then the default limits will be applied.

Reply:
Just like how it's done in `control_auto_velocity`

Reply:
True, we should be consistent. But what if setThrottleMax() was called in another method prior to switching the flight mode, would you then not explicitly want to set all the configurations again? So always outside of the if(gliding) loop?

Reply:
@sfuhrer The idea is that you need to be explicit with the control configuration. E..g everything that you don't set is automatically set back to default. So if another mode runs before you and your mode now runs and does not set anything, then automatically the configuration handler should set it back to default.
I hope that was your question.